src/test/groovy/com/intuit/graphql/orchestrator/integration/QueryDirectiveSpec.groovy

```diff
@@ -0,0 +1,59 @@
+package com.intuit.graphql.orchestrator.integration
+
+import com.google.common.collect.ImmutableMap
+import com.intuit.graphql.orchestrator.BooksService
+import com.intuit.graphql.orchestrator.GraphQLOrchestrator
+import com.intuit.graphql.orchestrator.PetsService
+import com.intuit.graphql.orchestrator.ServiceProvider
+import com.intuit.graphql.orchestrator.UserRestService
+import graphql.ExecutionInput
+import helpers.BaseIntegrationTestSpecification
+
+class QueryDirectiveSpec extends BaseIntegrationTestSpecification {
```

Comment:
Looks good but missing the validation of execution input for the downstream service.